### PR TITLE
simplify grammar

### PR DESCRIPTION
**Pull request by gkorland**

### CODE REVIEW

**Comment by MeirShpilraien** [comment]
We need to make sure those changes does not reduce performance.

**Comment by gkorland** [comment]
@MeirShpilraien do you expect any? I think it might improve performance.

File: src/jsonpath/grammer.pest [hunk 1]
```diff
@@ -44,20 +45,22 @@ term = _{select_term | decimal | string | boolean}
...
-all = {"*"}
+all = @{"*"}
 
-full_scan = {".."}
+full_scan = @{".."}
 
-bracket = _{numbers_range | numbers_list | string_list | all | "?" ~ filter}
+square_bracket = _{"[" ~ (numbers_range | numbers_list | string_list | all | "?" ~ filter) ~ "]"}
 
-element = _{"." ~ literal | full_scan ~ (literal | all | "[" ~ bracket ~ "]") | "." ~ all | ("."? ~ "[" ~ bracket ~ "]")}
```
Comment (MeirShpilraien) [inline src/jsonpath/grammer.pest:60]: I believe this was done for performance improvement where we assume most likeliy we will encounter `.literal` and the parser will match it faster.

Comment (MeirShpilraien) [reply src/jsonpath/grammer.pest:60]: But not sure so we should check

File: src/jsonpath/grammer.pest [hunk 2]
```diff
@@ -44,20 +45,22 @@ term = _{select_term | decimal | string | boolean}
...
+element = _{(full_scan | ".") ~ (literal | all | square_bracket) | square_bracket}
+
+function = _{literal ~ "(" ~ (literal)? ~ ("," ~ literal)* ~ ")"}
```
Comment (MeirShpilraien) [inline src/jsonpath/grammer.pest:63]: Where do we use this? Or is it a preparation for functions support?

**Review by MeirShpilraien — COMMENTED** [review]
Small comments about performance.

File: src/jsonpath/grammer.pest (hunk 2 above)
Comment (gkorland) [reply src/jsonpath/grammer.pest:63]: my mistake, I forgot it in the code for future support

**Pull request closed without merging**